src/transformation/visitors/switch.ts

```diff
@@ -145,11 +166,24 @@ export const transformSwitchStatement: FunctionVisitor<ts.SwitchStatement> = (st
                 (clause, index) => index >= start && containsBreakOrReturn(clause.statements)
             );
 
-            // Combine the default and all fallthrough statements
-            const defaultStatements: lua.Statement[] = [];
-            clauses
-                .slice(start, end >= 0 ? end + 1 : undefined)
-                .forEach(c => defaultStatements.push(...context.transformStatements(c.statements)));
+            const {
+                statements: defaultStatements,
+                hoistedStatements,
+                hoistedIdentifiers,
+            } = separateHoistedStatements(context, context.transformStatements(clauses[start].statements));
+
+            // Only push hoisted statements if this is the first time we're transforming the default clause
+            if (!defaultTransformed) {
+                prefixStatements.push(...hoistedStatements);
+                prefixIdentifiers.push(...hoistedIdentifiers);
+            }
+
+            // Combine the fallthrough statements
+            for (const clause of clauses.slice(start + 1, end >= 0 ? end + 1 : undefined)) {
+                let statements = context.transformStatements(clause.statements);
+                ({ statements } = separateHoistedStatements(context, statements));
```

Comment:
what if there are hoisted statements in here, are they ignored? if so, where are those included?

Reply:
These are the duplicate clauses added to the end of the final default when it fell through into others. Re-adding their hoisted statements a 2nd time would generate bad lua. I've added a comment to explain this is intentional.